dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/DefaultTrackerOwnershipManager.java

```diff
@@ -134,31 +140,37 @@ public void transferOwnership(
       Program program,
       OrganisationUnit orgUnit,
       boolean skipAccessValidation,
-      boolean createIfNotExists) {
+      boolean createIfNotExists)
+      throws ForbiddenException {
     if (entityInstance == null || program == null || orgUnit == null) {
       return;
     }
 
     User currentUser = userService.getUserByUsername(CurrentUserUtil.getCurrentUsername());
 
     if (hasAccess(currentUser, entityInstance, program) || skipAccessValidation) {
+      if (!programService.hasOrgUnit(program, orgUnit)) {
+        throw new ForbiddenException(
+            String.format(
+                "The program %s is not associated to the org unit %s",
+                program.getUid(), orgUnit.getUid()));
+      }
+
       TrackedEntityProgramOwner teProgramOwner =
           trackedEntityProgramOwnerService.getTrackedEntityProgramOwner(
               entityInstance.getId(), program.getId());
 
-      if (teProgramOwner != null) {
-        if (!teProgramOwner.getOrganisationUnit().equals(orgUnit)) {
-          ProgramOwnershipHistory programOwnershipHistory =
-              new ProgramOwnershipHistory(
-                  program,
-                  entityInstance,
-                  teProgramOwner.getOrganisationUnit(),
-                  teProgramOwner.getLastUpdated(),
-                  teProgramOwner.getCreatedBy());
-          programOwnershipHistoryService.addProgramOwnershipHistory(programOwnershipHistory);
-          trackedEntityProgramOwnerService.updateTrackedEntityProgramOwner(
-              entityInstance, program, orgUnit);
-        }
+      if (teProgramOwner != null && !teProgramOwner.getOrganisationUnit().equals(orgUnit)) {
```

Comment:
I just realized something regarding this merging of conditions here. 

Isnt the below `else if (createIfNotExists)` applicable only when  `teProgramOwner!=null` and should not be applicable even if `!teProgramOwner.getOrganisationUnit().equals(orgUnit)`

Or am I misunderstanding something?

Reply:
you're right, I removed that flag from master since it's always set to false, but I didn't remove it in the back ports, nice catch!